src/ContactMessageForm.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_contact_forms;
+
+use Drupal\contact\MessageForm;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Form controller for contact message forms.
+ *
+ * @internal
+ */
+class ContactMessageForm extends MessageForm {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function form(array $form, FormStateInterface $form_state): array {
+    $form = parent::form($form, $form_state);
+    /** @var \Drupal\contact\MessageInterface $contact_message */
+    $message = $this->entity;
+    /** @var \Drupal\contact\ContactFormInterface $contact_form */
+    $contact_form = $message->getContactForm();
+
+    $header = $contact_form->getThirdPartySetting('oe_contact_forms', 'header', FALSE);
+
+    if (!empty($header)) {
+      $form['header'] = [
+        '#markup' => $header,
+      ];
+    }
+
+    // Checkbox to accept privacy policy configured in the ContactForm.
+    $form['privacy_policy'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Privacy policy'),
+      '#description' => $contact_form->getThirdPartySetting('oe_contact_forms', 'privacy_policy', ''),
+      '#required' => TRUE,
+    ];
+
+    // Hide the optional fields the form manager has not configured to be used.
+    $optional_selected = $contact_form->getThirdPartySetting('oe_contact_forms', 'optional_fields', ['oe_country_residence', 'oe_telephone']);
```

Comment:
Is there a reason to default to those two field names? If not fields are selected, they should not be there.

Reply:
Fixed